p2p/simulations/network.go

```diff
@@ -630,10 +633,29 @@ func (n *Node) MarshalJSON() ([]byte, error) {
 	}{
 		Info:   n.NodeInfo(),
 		Config: n.Config,
-		Up:     n.Up,
+		Up:     n.Up(),
 	})
 }
 
+// UnmarshalJSON implements json.Unmarshaler interface so that we don't lose
+// Node.up status. IMPORTANT: The implementation is incomplete; we lose
```

Comment:
@frncmx can you give an example where `p2p.NodeInfo` (de)serialisation is actually needed? to the best of my understanding the info in `p2p.NodeInfo` should be reconstructed from the node info in the snapshot when bootstrapping the simulation

Reply:
I have no use case. But in general I would expect `MarshalJSON` and `UnmarshalJSON` to be symmetric. Thus I expect not to lose date during desalinization.

Reply:
I'm not sure why `p2p.NodeInfo` should be serialised in the first place. when we bootstrap the simulation we should not rely on deserialisation to construct our objects but rather pass the relevant data to the inferred constructors. makes sense?

Reply:
That makes sense. For me it can go either way; dropping `p2p.NodeInfo` is even better.
I'll have to check though where ewe might use `NodeInfo`... my initial guess it was there only for humans.

Reply:
@justelad this PR is aims to help progress on getting a race detector CI job in place (https://github.com/ethersphere/go-ethereum/issues/741)

To keep focus, https://github.com/ethersphere/go-ethereum/issues/1177 won't be fixed in this PR if that's OK for you. So if something breaks that won't block his PR.

Reply:
sure 👍